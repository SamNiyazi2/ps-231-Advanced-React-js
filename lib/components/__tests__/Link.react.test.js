
// 08/01/2020 10:37 pm - SSN - [20200801-2201] - [002] - Jest - Documenation training

import React from 'react';
import Link from '../Link.react';
import renderer from 'react-test-renderer';


test('Link changes the class when hovered', () => {


    const component = renderer.create(
        <Link page="http://bing.com">Bing</Link>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();



    // Manually trigger the callback
    tree.props.onMouseEnter();

    // Re-rendering
    component.toJSON();
    expect(tree).toMatchSnapshot();



    // Manually trigger the callback
    tree.props.onMouseLeave();
    
    // Re-render
    component.toJSON();
    expect(tree).toMatchSnapshot();



});

