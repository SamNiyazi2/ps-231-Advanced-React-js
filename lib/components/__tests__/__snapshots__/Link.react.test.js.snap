// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://bing.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noreferrer"
  target="_blank"
>
  Bing
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="normal"
  href="http://bing.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noreferrer"
  target="_blank"
>
  Bing
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://bing.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noreferrer"
  target="_blank"
>
  Bing
</a>
`;
